
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <meta name="author" content="Connectivity Map">
      
      <link rel="shortcut icon" href="../img/favicon.ico">
      <meta name="generator" content="mkdocs-1.1.2, mkdocs-material-6.1.7">
    
    
      
        <title>Running SigTools in MATLAB - CMap SigTool Documentation</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.19753c6b.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.196e0c26.min.css">
        
      
    
    
    
      
        
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>body,input{font-family:"Roboto",-apple-system,BlinkMacSystemFont,Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono",SFMono-Regular,Consolas,Menlo,monospace}</style>
      
    
    
    
    
      
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="none" data-md-color-accent="none">
      
  
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#pre-requisites-for-the-demo" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid" aria-label="Header">
    <a href=".." title="CMap SigTool Documentation" class="md-header-nav__button md-logo" aria-label="CMap SigTool Documentation">
      
  <img src="../img/banner-logo.svg" alt="logo">

    </a>
    <label class="md-header-nav__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header-nav__title" data-md-component="header-title">
      
        <div class="md-header-nav__ellipsis">
          <span class="md-header-nav__topic md-ellipsis">
            CMap SigTool Documentation
          </span>
          <span class="md-header-nav__topic md-ellipsis">
            
              Running SigTools in MATLAB
            
          </span>
        </div>
      
    </div>
    
      <label class="md-header-nav__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" data-md-state="active" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <button type="reset" class="md-search__icon md-icon" aria-label="Clear" data-md-component="search-reset" tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header-nav__source">
        
<a href="https://github.com/cmap/cmap-sig-tools" title="Go to repository" class="md-source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 00-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 01-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 000 40.81l195.61 195.6a28.86 28.86 0 0040.8 0l194.69-194.69a28.86 28.86 0 000-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    cmap/cmap-sig-tools
  </div>
</a>
      </div>
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="CMap SigTool Documentation" class="md-nav__button md-logo" aria-label="CMap SigTool Documentation">
      
  <img src="../img/banner-logo.svg" alt="logo">

    </a>
    CMap SigTool Documentation
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/cmap/cmap-sig-tools" title="Go to repository" class="md-source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 00-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 01-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 000 40.81l195.61 195.6a28.86 28.86 0 0040.8 0l194.69-194.69a28.86 28.86 0 000-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    cmap/cmap-sig-tools
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href=".." class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
    <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2" >
    <label class="md-nav__link" for="nav-2">
      Command Reference
      <span class="md-nav__icon md-icon"></span>
    </label>
    <nav class="md-nav" aria-label="Command Reference" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        <span class="md-nav__icon md-icon"></span>
        Command Reference
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../command-reference/shared_arguments/" class="md-nav__link">
      Shared arguments
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
    <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-2" type="checkbox" id="nav-2-2" >
    <label class="md-nav__link" for="nav-2-2">
      Connectivity analysis
      <span class="md-nav__icon md-icon"></span>
    </label>
    <nav class="md-nav" aria-label="Connectivity analysis" data-md-level="2">
      <label class="md-nav__title" for="nav-2-2">
        <span class="md-nav__icon md-icon"></span>
        Connectivity analysis
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../command-reference/sig_queryl1k_tool/" class="md-nav__link">
      queryl1k
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../command-reference/sig_gseapreranked_tool/" class="md-nav__link">
      gseapreranked
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../command-reference/sig_curie_tool/" class="md-nav__link">
      curie
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../command-reference/sig_introspect_tool/" class="md-nav__link">
      introspect
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../command-reference/sig_recall_tool/" class="md-nav__link">
      recall
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../command-reference/sig_gutc_tool/" class="md-nav__link">
      gutc
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
    <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-3" type="checkbox" id="nav-2-3" >
    <label class="md-nav__link" for="nav-2-3">
      Expression analysis
      <span class="md-nav__icon md-icon"></span>
    </label>
    <nav class="md-nav" aria-label="Expression analysis" data-md-level="2">
      <label class="md-nav__title" for="nav-2-3">
        <span class="md-nav__icon md-icon"></span>
        Expression analysis
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../command-reference/sig_marker_tool/" class="md-nav__link">
      marker
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../command-reference/sig_pca_tool/" class="md-nav__link">
      pca
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../command-reference/sig_tsne_tool/" class="md-nav__link">
      tsne
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
    <input class="md-nav__toggle md-toggle" data-md-toggle="nav-2-4" type="checkbox" id="nav-2-4" >
    <label class="md-nav__link" for="nav-2-4">
      Utilities
      <span class="md-nav__icon md-icon"></span>
    </label>
    <nav class="md-nav" aria-label="Utilities" data-md-level="2">
      <label class="md-nav__title" for="nav-2-4">
        <span class="md-nav__icon md-icon"></span>
        Utilities
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../command-reference/sig_annotate_tool/" class="md-nav__link">
      annotate
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../command-reference/sig_slice_tool/" class="md-nav__link">
      slice
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../command-reference/sig_collate_tool/" class="md-nav__link">
      collate
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../command-reference/sig_getgenesets_tool/" class="md-nav__link">
      getgenesets
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../command-reference/sig_score2rank_tool/" class="md-nav__link">
      score2rank
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../command-reference/sig_testmlr_tool/" class="md-nav__link">
      testmlr
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../command-reference/sig_trainmlr_tool/" class="md-nav__link">
      trainmlr
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../command-reference/sig_zscore_tool/" class="md-nav__link">
      zscore
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
    <input class="md-nav__toggle md-toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3" checked>
    <label class="md-nav__link" for="nav-3">
      Tutorials
      <span class="md-nav__icon md-icon"></span>
    </label>
    <nav class="md-nav" aria-label="Tutorials" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        <span class="md-nav__icon md-icon"></span>
        Tutorials
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Running SigTools in MATLAB
        <span class="md-nav__icon md-icon"></span>
      </label>
    
    <a href="./" class="md-nav__link md-nav__link--active">
      Running SigTools in MATLAB
    </a>
    
      
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#pre-requisites-for-the-demo" class="md-nav__link">
    Pre-requisites for the demo
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#connectivity-analysis-using-sigtools" class="md-nav__link">
    Connectivity analysis using SigTools
  </a>
  
    <nav class="md-nav" aria-label="Connectivity analysis using SigTools">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-running-a-cmap-query-against-an-l1000-dataset-using-the-queryl1k-tool" class="md-nav__link">
    1. Running a Cmap Query against an L1000 dataset using the QueryL1k tool
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-testing-enrichment-of-user-defined-sets-using-the-gsea-preranked-tool" class="md-nav__link">
    2. Testing enrichment of user-defined sets using the GSEA Preranked tool
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-querying-cell-viability-data-withe-the-curie-tool-using-cell-sets" class="md-nav__link">
    3. Querying cell viability data withe the Curie tool using cell-sets
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../docker_demo/" class="md-nav__link">
      Running SigTools with Docker
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../faq/" class="md-nav__link">
      FAQ
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#pre-requisites-for-the-demo" class="md-nav__link">
    Pre-requisites for the demo
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#connectivity-analysis-using-sigtools" class="md-nav__link">
    Connectivity analysis using SigTools
  </a>
  
    <nav class="md-nav" aria-label="Connectivity analysis using SigTools">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-running-a-cmap-query-against-an-l1000-dataset-using-the-queryl1k-tool" class="md-nav__link">
    1. Running a Cmap Query against an L1000 dataset using the QueryL1k tool
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-testing-enrichment-of-user-defined-sets-using-the-gsea-preranked-tool" class="md-nav__link">
    2. Testing enrichment of user-defined sets using the GSEA Preranked tool
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-querying-cell-viability-data-withe-the-curie-tool-using-cell-sets" class="md-nav__link">
    3. Querying cell viability data withe the Curie tool using cell-sets
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                  <h1>Running SigTools in MATLAB</h1>
                
                <hr />
<p>SigTools are command-line programs that allow easy execution of core CMap algorithms and methods via a standardized user interface. </p>
<p>This tutorial demonstrates running connectivity analyses on custom datasets in MATLAB using the source code available 
in the <a href="https://github.com/cmap/cmapM">cmapM</a> Github repository.</p>
<p>We also provide Docker images of the SigTools via DockerHub. 
These images come pre-configured with all the software dependencies for executing the tool on a local environment without 
requiring a commercial MATLAB license. To learn how to use the Docker images see <a href="../docker_demo/">here</a>.</p>
<h3 id="pre-requisites-for-the-demo">Pre-requisites for the demo<a class="headerlink" href="#pre-requisites-for-the-demo" title="Permanent link">&para;</a></h3>
<ul>
<li>Clone the cmapM code repository
<div class="highlight"><pre><span></span><code>git clone https://github.com/cmap/cmapM
</code></pre></div></li>
<li>Configure the MATLAB environment and download test data for the demo
<div class="highlight"><pre><span></span><code><span class="c">% within a MATLAB sesssion type:</span>
<span class="n">cd</span> <span class="s">cmapM</span>
<span class="n">setup</span>
</code></pre></div></li>
<li>Run all the demos, described in more detail below
<div class="highlight"><pre><span></span><code><span class="n">sig_tool_demos</span>
</code></pre></div></li>
</ul>
<h3 id="connectivity-analysis-using-sigtools">Connectivity analysis using SigTools<a class="headerlink" href="#connectivity-analysis-using-sigtools" title="Permanent link">&para;</a></h3>
<h4 id="1-running-a-cmap-query-against-an-l1000-dataset-using-the-queryl1k-tool">1. Running a Cmap Query against an L1000 dataset using the QueryL1k tool<a class="headerlink" href="#1-running-a-cmap-query-against-an-l1000-dataset-using-the-queryl1k-tool" title="Permanent link">&para;</a></h4>
<p>The QueryL1k tool computes a set-based enrichment similarity between input genesets (aka 
queries) and a small subset of L1000 perturbational gene-expression signatures. 
(Note that while the tool is optimized for datasets generated by the L1000 platform, 
any perturbational dataset can be used).</p>
<p>The algorithm operates as follows. First raw similarity (connectivity) scores 
between a query and CMap signatures are computed. While query methodology is 
agnostic to the specific similarity metric, the default choice is a non-parametric, two-tailed weighted gene-set enrichment score (Subramanian, A. et al. Cell 2017).</p>
<p>The raw scores are then scaled (normalized) by the signed-means to allow for 
comparisons across different queries.</p>
<p>Finally the statistical significance of the connections adjusted for multiple 
hypotheses is estimated. FDR q-values are estimated by comparing the 
distributions of treatments to null signatures in the dataset.</p>
<table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25</pre></div></td><td class="code"><div class="highlight"><pre><span></span><code><span class="n">DATASET_PATH</span> <span class="p">=</span> <span class="n">fullfile</span><span class="p">(</span><span class="n">cmapmpath</span><span class="p">,</span> <span class="s">&#39;demo-datasets&#39;</span><span class="p">);</span>

<span class="c">% Queries</span>
<span class="n">UP_GENESET</span> <span class="p">=</span> <span class="n">fullfile</span><span class="p">(</span><span class="n">DATASET_PATH</span><span class="p">,</span> <span class="s">&#39;queries/genesets/dexamethasone_resistance_up.gmt&#39;</span><span class="p">);</span>
<span class="n">DOWN_GENESET</span> <span class="p">=</span> <span class="n">fullfile</span><span class="p">(</span><span class="n">DATASET_PATH</span><span class="p">,</span> <span class="s">&#39;/queries/genesets/dexamethasone_resistance_down.gmt&#39;</span><span class="p">);</span>

<span class="c">% Gene Expression Dataset</span>
<span class="c">% Differential expression score matrix</span>
<span class="n">SCORE_FILE</span> <span class="p">=</span> <span class="n">fullfile</span><span class="p">(</span><span class="n">DATASET_PATH</span><span class="p">,</span> <span class="s">&#39;/l1000/m2.subset.10k/level5_modz.bing_n10000x10174.gctx&#39;</span><span class="p">);</span>
<span class="c">% Corresponding rank matrix</span>
<span class="n">RANK_FILE</span> <span class="p">=</span> <span class="n">fullfile</span><span class="p">(</span><span class="n">DATASET_PATH</span><span class="p">,</span> <span class="s">&#39;l1000/m2.subset.10k/rank.bing_n10000x10174.gctx&#39;</span><span class="p">);</span>
<span class="c">% Signature annotations</span>
<span class="n">SIG_META_FILE</span> <span class="p">=</span> <span class="n">fullfile</span><span class="p">(</span><span class="n">DATASET_PATH</span><span class="p">,</span> <span class="s">&#39;l1000/m2.subset.10k/siginfo.txt&#39;</span><span class="p">);</span>
<span class="c">% results folder</span>
<span class="n">OUT_PATH</span> <span class="p">=</span> <span class="s">&#39;results/queryl1k&#39;</span><span class="p">;</span>

<span class="c">% Run the queryl1k tool</span>
<span class="n">sig_queryl1k_tool</span><span class="p">(</span><span class="s">&#39;up&#39;</span><span class="p">,</span> <span class="n">UP_GENESET</span><span class="p">,</span><span class="c">...</span>
    <span class="s">&#39;down&#39;</span><span class="p">,</span> <span class="n">DOWN_GENESET</span><span class="p">,</span><span class="c">...</span>
    <span class="s">&#39;score&#39;</span><span class="p">,</span> <span class="n">SCORE_FILE</span><span class="p">,</span><span class="c">...</span>
    <span class="s">&#39;rank&#39;</span><span class="p">,</span> <span class="n">RANK_FILE</span><span class="p">,</span><span class="c">...</span>
    <span class="s">&#39;sig_meta&#39;</span><span class="p">,</span> <span class="n">SIG_META_FILE</span><span class="p">,</span><span class="c">...</span>
    <span class="s">&#39;max_col&#39;</span><span class="p">,</span> <span class="mi">50000</span><span class="p">,</span><span class="c">...</span>
    <span class="s">&#39;create_subdir&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span><span class="c">...</span>
    <span class="s">&#39;out&#39;</span><span class="p">,</span> <span class="n">OUT_PATH</span><span class="p">)</span>
</code></pre></div>
</td></tr></table>

<p><strong>Outputs:</strong> the tool produces the following output (in the <code>results</code> folder)</p>
<p><code>arfs/</code>: Per-query analysis report files (ARFs)</p>
<p><code>&lt;QUERY_NAME&gt;/query_result.gct</code> : a GCT format text file listing the annotations, 
connectivity scores and q-values for each signature in the dataset. The 
following fields are computed by the query tool:</p>
<ul>
<li><code>raw_cs</code> : Raw connectivity scores</li>
<li><code>norm_cs</code> : Normalized connectivity score computed by dividing the raw 
connectivity scores by the signed-mean scores of signatures (specified by the 
is_ncs_sig field in the signature metadata file) If the ncs_group field is not 
empty the scores are normalized within each group, otherwise the scores are 
normalized using the global means across all signatures.</li>
<li><code>fdr_q_nlog10</code> : Negative log10 transformed FDR q-values estimated relative to 
the null signatures (specified by the <code>is_null_sig</code> field in the signature 
annotation file).</li>
</ul>
<p><code>matrices/query</code> : Query parameters and result matrices in GCTx format for all 
queries:</p>
<ul>
<li><code>up.gmt</code>, <code>dn.gmt</code>: query genesets in GMT format</li>
<li><code>cs.gctx</code> : Raw connectivity scores matrix [signatures x queries] </li>
<li><code>ncs.gctx</code> : Normalized connectivity score matrix [signatures x queries]</li>
<li><code>fdr_qvalue.gctx</code> : Estimated false discovery rate q-values [signatures x 
queries]</li>
</ul>
<h4 id="2-testing-enrichment-of-user-defined-sets-using-the-gsea-preranked-tool">2. Testing enrichment of user-defined sets using the GSEA Preranked tool<a class="headerlink" href="#2-testing-enrichment-of-user-defined-sets-using-the-gsea-preranked-tool" title="Permanent link">&para;</a></h4>
<p>The GSEA Preranked tool computes set-based enrichment analysis against a user-defined 
rank-ordered dataset.  It determines whether a priori defined sets show 
statistically significant enrichment at either end of the ranking.</p>
<table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15</pre></div></td><td class="code"><div class="highlight"><pre><span></span><code><span class="c">% Signature sets</span>
<span class="n">SIG_SET</span> <span class="p">=</span> <span class="n">fullfile</span><span class="p">(</span><span class="n">DATASET_PATH</span><span class="p">,</span> <span class="s">&#39;queries/sigsets/cmap_sig_sets_n9380.gmt&#39;</span><span class="p">);</span>
<span class="c">% Set annotations</span>
<span class="n">QUERY_META_FILE</span> <span class="p">=</span> <span class="n">fullfile</span><span class="p">(</span><span class="n">DATASET_PATH</span><span class="p">,</span> <span class="s">&#39;queries/sigsets/cmap_sig_sets_info.txt&#39;</span><span class="p">);</span>
<span class="c">% Query result scores (Normalized connectivities)</span>
<span class="n">SCORE_FILE</span> <span class="p">=</span> <span class="n">fullfile</span><span class="p">(</span><span class="n">DATASET_PATH</span><span class="p">,</span> <span class="s">&#39;pre_computed_results/queryl1k/science_queries/arfs/DEX/query_result.gct&#39;</span><span class="p">);</span>
<span class="c">% results folder</span>
<span class="n">OUT_PATH</span> <span class="p">=</span> <span class="s">&#39;results/gseapreranked&#39;</span><span class="p">;</span>

<span class="c">% Run the gseapreranked tool</span>
<span class="n">sig_gseapreranked_tool</span><span class="p">(</span><span class="s">&#39;up&#39;</span><span class="p">,</span> <span class="n">SIG_SET</span><span class="p">,</span><span class="c">...</span>
    <span class="s">&#39;score&#39;</span><span class="p">,</span> <span class="n">SCORE_FILE</span><span class="p">,</span><span class="c">...</span>
    <span class="s">&#39;query_meta&#39;</span><span class="p">,</span> <span class="n">QUERY_META_FILE</span><span class="p">,</span><span class="c">...</span>
    <span class="s">&#39;create_subdir&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span><span class="c">...</span>
    <span class="s">&#39;out&#39;</span><span class="p">,</span> <span class="n">OUT_PATH</span><span class="p">);</span>
</code></pre></div>
</td></tr></table>

<h4 id="3-querying-cell-viability-data-withe-the-curie-tool-using-cell-sets">3. Querying cell viability data withe the Curie tool using cell-sets<a class="headerlink" href="#3-querying-cell-viability-data-withe-the-curie-tool-using-cell-sets" title="Permanent link">&para;</a></h4>
<p>The Curie tool computes a set-based enrichment similarity between input cell-line 
sets (aka queries) and a perturbational cell-fitness signature dataset. Note that while 
the tool is optimized for datasets generated by the PRISM platform, any high-dimensional
cell-fitness dataset can be used.</p>
<table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19</pre></div></td><td class="code"><div class="highlight"><pre><span></span><code><span class="c">% Queries (Cell line sets)</span>
<span class="n">CELL_SET</span> <span class="p">=</span> <span class="n">fullfile</span><span class="p">(</span><span class="n">DATASET_PATH</span><span class="p">,</span> <span class="s">&#39;queries/cellsets/curated_cellsets_n5.gmt&#39;</span><span class="p">);</span>

<span class="c">% Cell viability data</span>
<span class="c">% Cell fitness scores</span>
<span class="n">SCORE_FILE</span> <span class="p">=</span> <span class="n">fullfile</span><span class="p">(</span><span class="n">DATASET_PATH</span><span class="p">,</span> <span class="s">&#39;prism/repurposing-secondary/level5_modz.zspc.sqrtaud_n1258x489.gctx&#39;</span><span class="p">);</span>
<span class="c">% Corresponding rank matrix</span>
<span class="n">RANK_FILE</span> <span class="p">=</span> <span class="n">fullfile</span><span class="p">(</span><span class="n">DATASET_PATH</span><span class="p">,</span> <span class="s">&#39;prism/repurposing-secondary/rank_n1258x489.gctx&#39;</span><span class="p">);</span>
<span class="c">% Signature annotations</span>
<span class="n">SIG_META_FILE</span> <span class="p">=</span> <span class="n">fullfile</span><span class="p">(</span><span class="n">DATASET_PATH</span><span class="p">,</span> <span class="s">&#39;prism/repurposing-secondary/siginfo.txt&#39;</span><span class="p">);</span>

<span class="c">% Run the curie tool</span>
<span class="n">sig_curie_tool</span><span class="p">(</span><span class="s">&#39;es_tail&#39;</span><span class="p">,</span> <span class="s">&#39;up&#39;</span><span class="p">,</span><span class="c">...</span>
    <span class="s">&#39;up&#39;</span><span class="p">,</span> <span class="n">CELL_SET</span><span class="p">,</span><span class="c">...</span>
    <span class="s">&#39;score&#39;</span><span class="p">,</span> <span class="n">SCORE_FILE</span><span class="p">,</span><span class="c">...</span>
    <span class="s">&#39;rank&#39;</span><span class="p">,</span> <span class="n">RANK_FILE</span><span class="p">,</span><span class="c">...</span>
    <span class="s">&#39;sig_meta&#39;</span><span class="p">,</span> <span class="n">SIG_META_FILE</span><span class="p">,</span><span class="c">...</span>
    <span class="s">&#39;create_subdir&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span><span class="c">...</span>
    <span class="s">&#39;out&#39;</span><span class="p">,</span> <span class="n">OUT_PATH</span><span class="p">);</span>
</code></pre></div>
</td></tr></table>
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid" aria-label="Footer">
        
          <a href="../command-reference/sig_zscore_tool/" class="md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-footer-nav__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
            </div>
            <div class="md-footer-nav__title">
              <div class="md-ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                zscore
              </div>
            </div>
          </a>
        
        
          <a href="../docker_demo/" class="md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-footer-nav__title">
              <div class="md-ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Running SigTools with Docker
              </div>
            </div>
            <div class="md-footer-nav__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2020 Connectivity Map
          </div>
        
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/vendor.0ac82a11.min.js"></script>
      <script src="../assets/javascripts/bundle.f81dfb4d.min.js"></script><script id="__lang" type="application/json">{"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing"}</script>
      
      <script>
        app = initialize({
          base: "..",
          features: [],
          search: Object.assign({
            worker: "../assets/javascripts/worker/search.4ac00218.min.js"
          }, typeof search !== "undefined" && search)
        })
      </script>
      
    
  </body>
</html>